### PR TITLE
Support more capitalized string formats

### PR DESCRIPTION
**Pull request by ethanresnick**
First off, thanks so much for this library! It was truly wonderful to find this, already written and ready to be dropped in.

However, the code didn't work with a couple of my string formats (for example, PascalCase got an extra leading hyphen), so I updated the regex to be a bit more sophisticated and added tests for the newly supported formats. 

Hopefully these changes can get merged in. I'd love to go back to depending on your library directly rather than my fork!

### CODE REVIEW

**Comment by ethanresnick** [comment]
Hey Shahar, I just wanted to follow up on this.

Cheers,
Ethan

File: index.js [hunk 1]
```diff
@@ -24,9 +24,15 @@ var objectKeys = Object.keys || function (obj) {
...
-  });
+  var matches, words = [], wordsRe = /[A-Z]([A-Z]*(?![^A-Z])|[^A-Z]*)/g;
+
+  var pascalCaseString = str.charAt(0).toUpperCase() + str.slice(1);
+
+  while((matches = wordsRe.exec(pascalCaseString)) !== null) {
+    words.push(matches[0]);
+  }
+
+  return words.join("-").toLowerCase();
```
Comment (shahata) [inline index.js]: How about:
```js
return str.replace(/[A-Z]((?=[^A-Z])|[A-Z]*(?=[A-Z][^A-Z]|$))/g, function ($1) {
  return $1.toLowerCase();
}).replace(/^-/, '');
```

Comment (ethanresnick) [inline index.js]: Updated!

**Comment by shahata** [comment]
Thanks!

**Pull request closed without merging**